core/src/main/java/com/netflix/conductor/core/execution/tasks/SystemTaskWorkerCoordinator.java

```diff
@@ -75,15 +80,12 @@ public SystemTaskWorkerCoordinator(QueueDAO queueDAO, WorkflowExecutor workflowE
 
     @EventListener(ApplicationReadyEvent.class)
     public void initSystemTaskExecutor() {
-        int threadCount = properties.getSystemTaskWorkerThreadCount();
-        if (threadCount > 0) {
-            this.workflowSystemTasks.forEach(this::add);
-            this.systemTaskExecutor = new SystemTaskExecutor(queueDAO, workflowExecutor, properties, executionService);
-            new Thread(this::listen).start();
-            LOGGER.info("System Task Worker Coordinator initialized with poll interval: {}", pollInterval);
-        } else {
-            LOGGER.info("System Task Worker DISABLED");
-        }
+        int threadCount =
```

Comment:
Instead of this, the default value of 1 could be set in `ConductorProperties`.

Reply:
The default value is 10, this is to prevent the value being overridden to 0.

Reply:
If the value is set to `0`, shouldn't we use the default of `10`? The property's value determination gets complicated,

1. If the value is set to `0`, it becomes `1`.
2. If no value is set, then it's `10` which is the default.
3. If the value set is > 0, then the set value is used.

I vote to throw an error when the value is set to `0`. The error message could inform the user that to disable workers, the flag can be set as `false`.

Reply:
On a related note, it might be worth considering the default values a function of the available processors. Something like this,

```java
private int systemTaskWorkerThreadCount = Runtime.getRuntime().availableProcessors() * 2;
```